src/platform/native.zig

```diff
@@ -587,60 +587,81 @@ pub const Platform = struct {
 
 pub const BackingTimer = std.time.Timer;
 
+var app: App = undefined;
+
 pub fn main() !void {
     var gpa = std.heap.GeneralPurposeAllocator(.{}){};
     defer _ = gpa.deinit();
     const allocator = gpa.allocator();
 
-    var core = try Core.init(allocator);
-    defer core.internal.deinit();
-    var app: App = undefined;
-
-    try app.init(&core);
-    defer app.deinit(&core);
-
-    // Glfw specific: initialize the user pointer used in callbacks
-    core.internal.initCallback();
-
-    const window = core.internal.window;
-    while (!window.shouldClose()) {
-        if (core.internal.wait_event_timeout > 0.0) {
-            if (core.internal.wait_event_timeout == std.math.inf(f64)) {
-                // Wait for an event
-                try glfw.waitEvents();
-            } else {
-                // Wait for an event with a timeout
-                try glfw.waitEventsTimeout(core.internal.wait_event_timeout);
-            }
+    var core = try core_init(allocator);
+    defer core_deinit(core);
+
+    try app.init(core);
+    defer app.deinit(core);
+
+    while (!core.internal.window.shouldClose()) {
+        try core_update(core, null);
+
+        try app.update(core);
+    }
+}
+
+pub fn core_init(allocator: std.mem.Allocator) !*Core {
+    const core: *Core = try allocator.create(Core);
+    core.* = try Core.init(allocator);
+
+    // // Glfw specific: initialize the user pointer used in callbacks
+    core.*.internal.initCallback();
+
+    return core;
+}
+
+pub export fn core_deinit(core: *Core) void {
+    core.internal.deinit();
+    // assumes that core.allocator is the same allocator used to allocate core
+    core.allocator.destroy(core); // "I used the core to destroy the core"
```

Comment:
How about we pass the allocator into this function instead? Would eliminate these semi-confusing comments, and is a better approach to use generally anyway.